graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/streams/StreamRuleType.java

```diff
@@ -22,24 +22,36 @@
  */
 package org.graylog2.plugin.streams;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
 public enum StreamRuleType {
-    EXACT(1),
-    GREATER(3),
-    SMALLER(4),
-    REGEX(2),
-    PRESENCE(5);
+    EXACT(1, "match exactly", "match exactly"),
+    REGEX(2, "match regular expression", "match regular expression"),
+    GREATER(3, "greater than", "be greater than"),
+    SMALLER(4, "smaller than", "be smaller than"),
+    PRESENCE(5, "field presence", "be present");
 
     private final int value;
+    private final String shortDesc;
+    private final String longDesc;
 
-    StreamRuleType(final int value) {
+    StreamRuleType(final int value, final String shortDesc, final String longDesc) {
         this.value = value;
+        this.shortDesc = shortDesc;
+        this.longDesc = longDesc;
     }
 
     public int toInteger() {
         return value;
     }
 
-    public static StreamRuleType fromInteger(final int numeric) {
+    public static StreamRuleType fromInteger(@JsonProperty("value") final int numeric) {
```

Comment:
What purpose does the `@JsonProperty` annotation serve here? It would only be relevant in a `@JsonCreator` method.